### DISCUSSION TITLE
Cursor changes color at the begining and end-of-line

### DISCUSSION THREAD

**Discussion by italomaia** [helix-editor/helix#10127] [General]
### Summary

I noticed that when using the "bar" cursor shape, the cursor color changes when the cursor is at the beginning or end of a line. 

### Reproduction Steps

 

I tried this:

1. Make sure cursor-shape is "bar"
2. `hx`
3. go into insert_mode
4. type "hello"

I expected this to happen: cursor should keep its color regardless of the position.

Instead, this happened: cursor changes color

I tried with different themes, including the default theme and had the same result. 

### Helix log

Nothing in the logs

### Platform

Linux

### Terminal Emulator

xfce4-terminal

### Installation Method

snap

### Helix Version

helix 24.3 (2cadec0b)

**Comment by the-mikedavis** [comment]
Can you attach a [asciicast](https://asciinema.org/) / gif / video?

**Reply by italomaia** [reply]
![bar](https://github.com/helix-editor/helix/assets/14670/037599dc-602c-465e-82ce-f25a972b2e38)
As can be seen above, the cursor changes color depending on its position when the shape is bar.

![block](https://github.com/helix-editor/helix/assets/14670/5d218e7e-a0df-4b33-a03f-626ce43ce163)
As can be seen above, the cursor keeps its color regardless of t he position.

**Comment by CptPotato** [comment]
If I remember correctly, the color of the bar cursor is controlled by the terminal emulator, not the application.

**Comment by rizo** [comment]
Thanks @CptPotato. I think this should be mentioned in the docs. In many themes it becomes impossible to see the cursor.
